internal/clients/http/gcp/gcp_client.go

```diff
@@ -131,31 +180,37 @@ func (c *gcpClient) InsertInstances(ctx context.Context, params *clients.GCPInst
 			NamePattern: params.NamePattern,
 			Count:       &amount,
 			MinCount:    &amount,
-			InstanceProperties: &computepb.InstanceProperties{
-				Labels: map[string]string{
-					"rhhcc-rid": uuid,
-				},
-				Disks: []*computepb.AttachedDisk{
-					{
-						InitializeParams: &computepb.AttachedDiskInitializeParams{
-							SourceImage: &params.ImageName,
-						},
-						AutoDelete: ptr.To(true),
-						Boot:       ptr.To(true),
-						Type:       ptr.To(computepb.AttachedDisk_PERSISTENT.String()),
-					},
-				},
-				MachineType: ptr.To(params.MachineType),
-				NetworkInterfaces: []*computepb.NetworkInterface{
-					{
-						Name: ptr.To("global/networks/default"),
+		},
+	}
+
+	if params.LaunchTemplateName != "" {
+		template := fmt.Sprintf("global/instanceTemplates/%s", params.LaunchTemplateName)
+		req.BulkInsertInstanceResourceResource.SourceInstanceTemplate = &template
+	} else {
+		req.BulkInsertInstanceResourceResource.InstanceProperties = &computepb.InstanceProperties{
+			Labels: map[string]string{
+				"rhhcc-rid": uuid,
+			},
+			Disks: []*computepb.AttachedDisk{
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						SourceImage: &params.ImageName,
 					},
+					AutoDelete: ptr.To(true),
+					Boot:       ptr.To(true),
+					Type:       ptr.To(computepb.AttachedDisk_PERSISTENT.String()),
 				},
-				Metadata: &computepb.Metadata{
-					Items: metadata,
+			},
+			MachineType: ptr.To(params.MachineType),
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				{
+					Name: ptr.To("global/networks/default"),
 				},
 			},
-		},
+			Metadata: &computepb.Metadata{
+				Items: metadata,
+			},
+		}
 	}
```

Comment:
We cannot just throw away users settings, at the moment the UI enforces users to set instance type, amount and pubkey. This code ignores the setting and only applies the template.

We can change this, but this will require:

* updating AWS backend to ignore user input
* updating UI to "grey out" required fields
* discuss this with UX

What I suggest at this point:

* investigate if you can use a template and still provide pubkey/instancetype/amount in the GCP API
* if this can be done easily, let’s do this similarly to AWS
* if not, let’s implement GCP this way and create an EPIC to do all the above for AWS

@ezr-ondrej ?

Reply:
Apologies for any confusion, I realized that I made an error by omitting the metadata section when the template name is present. 

It appears that we do consider the amount(line 181) and public key ID (after the upcoming change), the only missing piece is the **machine type**. 

But, I have noticed that when providing a template name, the created instances use the machine type that the user has chosen and not the machine type that belongs to the template. To address this, my suggestion is to grey out the machine type section if a machine type was provided in the chosen template.

That is also true for the image type, the image type was ovrriden as well and the instances were created from the the image the user has provide and not from the template's image.

Reply:
Problem is that with AWS you can create a launch template that is completely empty - does not define anything other than its name. If we want to go that route, we need to find out if selected template does define instance type, pubkey and other fields which are in question.

That is why we have decided that simply we would enforce the user the always provide all the info and it would always override templates. It is also more simple in the UI for now. At least for AWS, my proposal here is to do the same for GCP.

I suggest to do the same with GCP and we can create and track an epic that would implement what you are suggesting - a UI code that will "disable" some elements. API requests would have empty strings or zeroes for values which were not set.

Reply:
The GCP API seems to support overriding, can we do this similarly to AWS at least for now and if you don’t like this feel free to refactor the UI and add support for skipping options when template is set.

https://cloud.google.com/compute/docs/reference/rest/v1/instances/bulkInsert

_Specifies the instance template from which to create instances. You may combine sourceInstanceTemplate with instanceProperties to override specific values from an existing instance template. Bulk API follows the semantics of JSON Merge Patch [described by RFC 7396](https://www.ietf.org/rfc/rfc7396.txt)._

Reply:
Created a ticket to track this issue: https://issues.redhat.com/browse/HMS-1861